pkg/utils/validator/clusterresourceplacement_test.go

```diff
@@ -987,6 +987,31 @@ func TestValidateClusterResourcePlacement_PickNPlacementPolicy(t *testing.T) {
 			wantErr:    true,
 			wantErrMsg: "PropertySelector is not allowed for PreferredDuringSchedulingIgnoredDuringExecution affinity",
 		},
+		"invalid placement policy - PickN with invalid property sorter in PreferredDuringSchedulingIgnoredDuringExecution affinity": {
+			policy: &placementv1beta1.PlacementPolicy{
+				PlacementType:    placementv1beta1.PickNPlacementType,
+				NumberOfClusters: &positiveNumberOfClusters,
+				Affinity: &placementv1beta1.Affinity{
+					ClusterAffinity: &placementv1beta1.ClusterAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []placementv1beta1.PreferredClusterSelector{
+							{
+								Preference: placementv1beta1.ClusterSelectorTerm{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{"test-key1": "test-value1"},
+									},
+									PropertySorter: &placementv1beta1.PropertySorter{
+										Name:      "resources.kubernetes-fleet.io/total-cpu",
```

Comment:
what kinds name can this be? Will this cause unexpected error again?

Reply:
This PR handles the name case and discusses introducing user errors to prevent unexpected errors occurring due to user input https://github.com/Azure/fleet/pull/808